src/styles/main.scss

```diff
@@ -0,0 +1,76 @@
+@use 'variables' as *;
+
+$days: (
+  mon: 1,
+  tue: 2,
+  wed: 3,
+  thu: 4,
+  fri: 5,
+  sat: 6,
+  sun: 7,
+);
+
+body {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  min-height: 100vh;
+  margin: 0;
+  font-family: Arial, sans-serif;
+}
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: calc(#{$day-size} * 7 + $gap * 6 + $padding * 2);
+  padding: $padding;
+
+  &__day {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    border: $day-border solid black;
+    background-color: $default-color;
+    position: relative;
+    font-size: 30px;
+    font-family: Arial, sans-serif;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition:
+      background-color $transition-long,
+      transform $transition-long;
+
+    &::before {
+      content: counter(day);
+      counter-increment: day;
```

Comment:
The `counter` property is used here, but there is no `counter-reset` defined for `day`. Ensure you have a `counter-reset` property set somewhere to initialize the counter, or this might not work as expected.